src/Formatter/CurlCommandFormatter.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Http\Message\Formatter;
+
+use Http\Message\Formatter;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * A formatter that prints a cURL command for HTTP requests.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class CurlCommandFormatter implements Formatter
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function formatRequest(RequestInterface $request)
+    {
+        $command = sprintf('curl %s', escapeshellarg((string) $request->getUri()->withFragment('')));
+        if ($request->getProtocolVersion() === '1.0') {
+            $command .= ' --http1.0';
+        } elseif ($request->getProtocolVersion() === '2.0') {
+            $command .= ' --http2';
+        }
+
+        $method = strtoupper($request->getMethod());
+        if ('HEAD' === $method) {
+            $command .= ' --head';
+        } elseif ('GET' !== $method) {
+            $command .= ' --request '.$method;
+        }
+
+        $command .= $this->getHeadersAsCommandOptions($request);
+
+        $body = $request->getBody();
+        if ($body->getSize() > 0) {
+            if (!$body->isSeekable()) {
+                return 'Cant format Request as cUrl command if body stream is not seekable.';
+            }
+            $command .= sprintf(' --data %s', escapeshellarg($body->__toString()));
```

Comment:
i wonder if we should have some sort of sanity check to not put huge files in here? say if the body size > 5000 we truncate or just do something like [too large body] in the command? if we use this formatter eg in the symfony debug toolbar, it could break on large files